packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/WebViewExtendedApi.java

```diff
@@ -0,0 +1,11 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.webviewflutter;
+
+import androidx.annotation.Nullable;
+/** Define extending APIs for {@link android.webkit.WebView} */
+public interface WebViewExtendedApi {
```

Comment:
Instead of adding a new method to `WebView`, you should add a n`onScrollChanged` callback to the Dart `WebView`. The same way `WebViewClient` adds a callback: https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_android/lib/src/android_webview.dart#L759

I would also make the callback include the `WebView` for convenience:

```dart
final void Function(WebView webView, int left, int top, int oldLeft, int oldTop)? onScrollChanged;
```

Reply:
Do you mean `WebViewExtendedApi` is not necessarily needed? I did read how the `WebViewClient` and attempted to do the same but couldn't find another way to set a scroll listener to a `WebView` without defining the `WebViewExtendedApi `.

The difference between setting a `ScrollListener` and a `WebViewClient` is that the `webkit.WebView` expose the method `setWebViewClient` for public use regardless of the Android required API level, this is unlike `setOnScrollChangeListener` since this is the method required min API level 23. The workaround here is to override the method`onScrollChanged` of the `WebView` and get the offset changed event from there.

Do you have another approach for this?

Reply:
Yea you can start by adding the `onScrollChanged` parameter to `WebView`:

```dart
class WebView {
  WebView({this.useHybridComposition = false, this.onScrollChanged}) : super.detached() {
    api.createFromInstance(this);
  }

  final void Function(WebView webView, int left, int top, int oldLeft, int oldTop)? onScrollChanged;
}
```

Then will need to establish a Flutter API for WebView in the `pigeons/android_webview.dart` file:

```dart
@FlutterApi()
abstract class WebViewFlutterApi {
  void onScrollChanged(int identifier, ...);
}
```

Then you can run the generator and implement the Flutter API that is generated. The implementation goes in `lib/src/android_webview_api_impl.dart`.

Then in Java you'll need to override the `onScrollChanged` method in the `WebViewPlatformView` class found in `WebViewHostApiImpl.java` class. You can then instantiate a `WebViewFlutterApi` and call the `WebViewFlutterApi.onScrollChanged` method in this overridden method.  

I may have missed a few things in this explanation, so definitely feel free to ask me for more clarification. The plugin is still suffering from tech debt, so adding new features still requires more work than they will in the future. I can also contribute code to this PR if needed.

Reply:
Thank you for the feedback. I did as you said but even though it helped to make things simple I still end up needing to retain `WebViewExtendedApi `.